### DISCUSSION TITLE
MPD and Displayport/HDMI -- alsa says no device

### DISCUSSION THREAD

**Discussion by c2b2pss** [MusicPlayerDaemon/MPD#1420] [Q&A]
So I can get sound through the HDMI, but for some reason it is not native DSD. It shows as 48Khz ouput 2 ch. I have a receiver that native decodes DSD at high rates.

If I attach my USB DSD DAC it shows DSD as input and the audio is fine too.

1.  Is there any limitation on HDMI sample rate.  The file is 225 bits -352.8 KHz.  
2. How do I force native-dsd ouput on HDMI using mpd.conf.
3. Any way to check if my mpd has DSD support ??

Thanks!

mpd.conf file:

```
music_directory         "/media/music"

user                            "root"

group                           "root"

#bind_to_address                "~/.mpd/socket"

port                            "6600"

#log_level                      "default"

gapless_mp3_playback                    "yes"

restore_paused                           "yes"

save_absolute_paths_in_playlists        "yes"

gapless_mp3_playback                    "yes"

metadata_to_use "artist,album,title,track,name,genre,date,composer,performer,disc$

#follow_outside_symlinks        "yes"

#follow_inside_symlinks         "yes"

#zeroconf_enabled               "yes"

#password                        "password@read,add,control,admin"

#zeroconf_enabled               "yes"

#zeroconf_name                  "music"

#password                        "password@read,add,control,admin"

#volume_normalization           "no"

#filesystem_charset              "UTF-8"

#id3v1_encoding                  "ISO-8859-1"

realtime_option {
        memlock "yes"
        stack_reserve "1024"
        heap_reserve "10240"
        main_priority "OTHER:0"
        player_priority "FIFO:32"
        decoder_priority "FIFO:31"
        update_priority "OTHER:0"
}

zeroconf_enabled    "yes"
zeroconf_name "music-OptiPlex-790"
audio_output {
   type "alsa"
   name "HDMI 0"
   device "hw:0,3"
   buffer_time "40000"
   period_time "10000"
   auto_format  "no"
   auto_channels "no"
   mixer_type "none"
   replay_gain_handler "none"
   use_mmap "yes"
}
```

Weirdly alsa does not report the HDMI as input....maybe it is some thing at kernel level. With only HDMI no other dac....

```
aplay -Ll
null
    Discard all samples (playback) or generate zero samples (capture)
aplay: device_list:270: no soundcards found...
```
With USB DAC attached:

```
cat /proc/asound/card*
cat: /proc/asound/card0: Is a directory
cat: /proc/asound/card1: Is a directory
 0 [PCH            ]: HDA-Intel - HDA Intel PCH
                      HDA Intel PCH at 0xe1a60000 irq 26
 1 [D10s           ]: USB-Audio - D10s
                      Topping D10s at usb-0000:00:1d.0-1.2, high speed
```

**Comment by MPconta** [comment]
I'm sorry to disappoint you. As far as i know, there is absolutely no way to get DSD over HDMI (from a PC). I think there exists no kernel-driver and no driver for ALSA too.

**Comment by MaxKellermann** [comment]
> realtime_option {

If this doesn't cause MPD to abort startup with an error message, it means you're using a MPD fork with bad code that has been rejected by this project due to quality problems. Please do not ask for help with MPD forks. Only MPD from this code repository is supported.

**Comment by c2b2pss** [comment]
Max,

You have gotten a very wrong impression of where I am coming from. Let me apologize in any case for any slights. I fully understand the pain I myself have felt trying other peoples forks of this mpd.  And so I understand also how bad it is for you -- seriously bad -- some of the forks are as I have spent days non-stop trying to get any to work.  I hate it when things are just such badly written and then put out for consumption. The mpd forks are a bloody muddy mess!!! I just want a linux server to play my audio....and mpd as the main player, Period. The rest be damned, I say.  **I don't want to get involved in this name calling, please.**

I am seriously not interested in forking or interfering -- I am trying painfully to see if some "talked" about changes can be made to mpd . It is a personal venture...but when there is something basic like getting HDMI audio etc., that I am clueless about I hope to post and get some input.  IF I find something I know works, I am very happy to share here and just leave the credit to you.

If this is against the discussions here, I am sorry again, and I'll plod along on my own. 

Regards!

**Reply by MaxKellermann** [reply]
HDMI output works very well with MPD, but as @MPconta already said, DSD over HDMI is not possible. That is not a MPD restriction, but a HDMI restriction.

Of course, you can use any MPD fork you like - it's open source, and forking is the whole point of open source, that's good! There is no "name calling" here, it's just that this project can only support the programs published by this project, and we cannot support other people's programs.

**Reply by c2b2pss** [reply]
Yes, I heard about real-time kernel possibilities. I am using your clean code to see if I can experiment.  I wish I was not so damn audiophillic trying to get some pristine state, but thats me. In this specific issue, there is audio coming out of my USB DAC and analog. So yes it is working and mpd is doing its thing.  But there is no audio from HDMI. That's all there is to my issue.

The fact I can play at will through my DAC optical and analog means I know the changes I made are successful at least are in the right direction.  The question is about HDMI, which may be my system and nothing to do with mpd.

**Reply by c2b2pss** [reply]
> HDMI output works very well with MPD, but as @MPconta already said, DSD over HDMI is not possible. That is not a MPD restriction, but a HDMI restriction.
> 
> Of course, you can use any MPD fork you like - it's open source, and forking is the whole point of open source, that's good! There is no "name calling" here, it's just that this project can only support the programs published by this project, and we cannot support other people's programs.

Fully agree!!  Peace!

Yes, it has taken a lot of reading to know that DSD over HDMI is not the "HDMI" one thinks of.  I went hunting because my receiver manual says it takes DSD only over HDMI-- without any clarification.  Everything is like pulling teeth!.

**Reply by MaxKellermann** [reply]
> Yes, I heard about real-time kernel possibilities.

Oh. That explains the weird settings in your mpd.conf - you followed some bad advice that spreads around the internet faster than good advice.

Good advice is here: https://mpd.readthedocs.io/en/stable/user.html#real-time-scheduling

Repeat: if you don't experience xruns, real-time scheduling will have **no effect**. Real-time schuedling is **not** about audio quality, it's only about preventing xruns.

**Reply by c2b2pss** [reply]
Thank you, Max. Yes, I am beginning to have a lot of salt by my side when I read stuff.  Everything out there needs teaspoons of salt.

So I guess then what is the status of mpd handling DSD. Can it convert DSD to PCM or such magic? Where the hell does all this talk of mpd and DSD come from? Aaaargh....It's not for the faint of heart.

I just want to listen to a lot of my DSD music via mpd -- rather than spend $$$$$ on those commerical audiophile snakeoil projects.

**Reply by MaxKellermann** [reply]
> So I guess then what is the status of mpd handling DSD. Can it convert DSD to PCM or such magic?

Yes, and it's documented: https://mpd.readthedocs.io/en/stable/user.html#direct-stream-digital-dsd

> Where the hell does all this talk of mpd and DSD come from?

What talk?

**Reply by c2b2pss** [reply]
Aah...well, I missed that! 

It says "Convert DSD to PCM on-the-fly."  That solution is good enough for me.  What would I need to do to get that to work? That will end a lot of my angst  :-(

And will you mpd source code do that? Silly question, sorry.

**Reply by c2b2pss** [reply]
Is there a link for converting DSD onthe fly to PCM in mpd?

**Reply by MaxKellermann** [reply]
> What would I need to do to get that to work?

Nothing. It's the default mode if native DSD is not supported.

> Is there a link for converting DSD onthe fly to PCM in mpd?

What kind of link?

**Reply by c2b2pss** [reply]
Sorry I meant instructions -- like mpd.conf parameters or other settings? How do I tell it to convert all DSD to PCM on the fly?

**Reply by MaxKellermann** [reply]
Can you read?

> It's the default mode if native DSD is not supported.

And it's documented in https://mpd.readthedocs.io/en/stable/user.html#direct-stream-digital-dsd

**Reply by c2b2pss** [reply]
Ok. OK. I get it. Whew!  :-)

**Reply by c2b2pss** [reply]
Without malice ,why is there so much anxiety in all these discussions? It's very intimidating to ask for help, contrary to the professed sentiments of any such group.  Not all of us are coding geeks -- we just want to make sure info is correctly understood.

It is very stressful for lesser folk to navigate these .

**Reply by MaxKellermann** [reply]
It is very stressful when people keep repeating questions when the answer has already been given, twice.
It is also very stressful to invest a lot of free time to write documentation, and people ask questions without having read a single piece of it, and even when you point them directly at the relevant part of the documentation, they still don't read it and repeat the question.
You were not asking to make sure something is correctly understood, you kept repeating the same question without context and without any reference to previous replies.
And about ...
> why is there so much anxiety in all these discussions? It's very intimidating to ask for help

You ask why there is anxiety, when in the second sentence you say it's your own anxiety. I have no idea, but I I don't have time to discuss your anxiety. So let's stay on-topic and discuss MPD and nothing else.
